eth/downloader/testchain_test.go

```diff
@@ -107,33 +155,37 @@ func init() {
 	wg.Add(len(chains))
 	for _, chain := range chains {
 		go func(blocks []*types.Block) {
-			newTestBlockchain(blocks)
+			newTestBlockchain(blocks, isZk)
 			wg.Done()
 		}(chain.blocks[1:])
 	}
 	wg.Wait()
 
 	// Mark the chains pregenerated. Generating a new one will lead to a panic.
-	pregenerated = true
+	if isZk {
+		zkpregenerated = true
+	} else {
+		pregenerated = true
+	}
```

Comment:
The initChains function takes longer than you might think.
That variable exists to prevent calls to newTestBlockchain after the init function has finished, which makes sense.